core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/UnwrapCastInComparison.java

```diff
@@ -562,11 +562,17 @@ private static Instant getInstantWithTruncation(TimestampWithTimeZoneType type,
 
     public static Expression falseIfNotNull(Expression argument)
     {
-        return and(new IsNullPredicate(argument), new NullLiteral());
+        // Coerce NullLiteral to boolean type otherwise predicate pushdown rule could throw an error
```

Comment:
This comment is unnecessary. Expressions in the IR need to be properly typed, and NullLiteral doesn't have any type, so it's expected that a cast would be needed.